registration/registration-services/src/main/java/io/mosip/registration/service/packet/impl/PacketUploadServiceImpl.java

```diff
@@ -140,12 +140,15 @@ public ResponseDTO uploadSyncedPackets() {
 				batchCount);
 		if (syncedPackets != null && !syncedPackets.isEmpty()) {
 			for(Registration registration : syncedPackets) {
-				if(registration.getServerStatusCode().equals(RegistrationConstants.PACKET_STATUS_CODE_REREGISTER))
+				if(registration.getServerStatusCode() != null && registration.getServerStatusCode().equals(RegistrationConstants.PACKET_STATUS_CODE_REREGISTER))
```

Comment:
you could have changed this as below to avoid NPE
RegistrationConstants.PACKET_STATUS_CODE_REREGISTER.equals(registration.getServerStatusCode())

Reply:
Modified the condition as suggested.